eva-server/src/main/java/uk/ac/ebi/eva/server/repository/VariantEntityRepositoryImpl.java

```diff
@@ -1,20 +1,68 @@
+/*
+ * European Variation Archive (EVA) - Open-access database of all types of genetic
+ * variation data from all species
+ *
+ * Copyright 2015 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package uk.ac.ebi.eva.server.repository;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Sort;
 import org.springframework.data.mongodb.MongoDbFactory;
 import org.springframework.data.mongodb.core.MongoTemplate;
 import org.springframework.data.mongodb.core.convert.MappingMongoConverter;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+
+import uk.ac.ebi.eva.commons.models.metadata.VariantEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * VariantEntityRepository(Custom) implementation.
+ *
+ * Used due to custom DBObject to VariantEntity conversion, and custom region query using a margin for efficiency.
+ */
+public class VariantEntityRepositoryImpl implements VariantEntityRepositoryCustom {
```

Comment:
Again a bit more of clarification for those not so versed in Spring Data repositories:

> Concrete implementation of the VariantEntityRepository interface (relationship inferred by Spring), due to a custom DBObject to VariantEntity conversion
> \<p\>
> It also implements the VariantEntityRepositoryCustom interface, to provide an explicit implementation of the region query, using a margin for efficiency.